File: Stack/20. Valid Parentheses/solution.swift


class Solution {
    func isValid(_ s: String) -> Bool {
        
        var stack : [Character] = []
        
        let brakets : [Character : Character] = ["}":"{", "]":"[", ")" : "("]
        
        
        for c : Character in s {
            if brakets.values.contains(c) {
                stack.append(c)
            }
            else if brakets.keys.contains(c) {
                if stack.isEmpty || stack.popLast() != brakets[c] {
                    return false
                }
            }
        }
        
        return stack.count == 0
    }
}




let s = "()[]{}"

print(Solution().isValid(s))

File: Stack/20. Valid Parentheses/solution.py

class Solution:
    def isValid(self, s: str) -> bool:

        brakets = {"}": "{", "]":"[", ")":"("}

        stack = []

        for c in s:
            if c in brakets.values():
                stack.append(c)
            elif c in brakets.keys():
                if not stack or stack.pop() != brakets[c]:
                    return False
        
        return len(stack) == 0
    



if __name__ == '__main__':

    """ Example 1: """
    s = "()[]{}"
    print(Solution().isValid(s))